Generated APK with GeckoView kernel crashes on launch: NPE in GeckoRuntime$LifecycleListener.onResume

## Bug
APK built with WebToApp 2.5.0 using the GeckoView kernel crashes immediately on launch. The in-app preview (also GeckoView) works fine — only the generated shell APK crashes.

## Stack trace
```
java.lang.RuntimeException: Failed to call observer method
    at androidx.lifecycle.LifecycleRegistry.addObserver(SourceFile:249)
    at org.mozilla.geckoview.GeckoRuntime.init(SourceFile:375)
    at org.mozilla.geckoview.GeckoRuntime.create(SourceFile:12)
    at com.webtoapp.core.engine.GeckoViewEngine$Companion.createRuntime(SourceFile:187)
    ...
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.os.Handler.post(java.lang.Runnable)' on a null object reference
    at org.mozilla.geckoview.GeckoRuntime$LifecycleListener.onResume(SourceFile:65)
```

## Reproduction
1. Download GeckoView kernel in Browser Kernel page
2. Create web app, build APK with GeckoView engine
3. Install and launch generated APK → crash 100% on launch

## Environment
- WebToApp 2.5.0 (GitHub release)
- Android 16 (LineageOS 23.2, OnePlus CPH2465), arm64
- The crash looks like GeckoRuntime is created before/without the shell app initializing ProcessLifecycleOwner handler — the in-app preview path presumably initializes it, the generated shell does not.

--- a/app/src/main/java/com/webtoapp/core/engine/GeckoViewEngine.kt
+++ b/app/src/main/java/com/webtoapp/core/engine/GeckoViewEngine.kt
@@ -315,7 +315,33 @@ class GeckoViewEngine(
             else -> "\"" + value.toString().replace("\\", "\\\\").replace("\"", "\\\"") + "\""
         }
 
+        /**
+         * GeckoView 142 race workaround (#582). GeckoRuntime.init() calls
+         * ProcessLifecycleOwner.addObserver(LifecycleListener) right after
+         * GeckoThread.launch(). When the process lifecycle is already RESUMED at that
+         * moment (generated apps create the engine from Compose's first layout, after
+         * the activity has resumed), LifecycleRegistry synchronously dispatches
+         * ON_RESUME and LifecycleListener.onResume() dereferences
+         * ThreadUtils.sGeckoHandler — a field the native side only assigns later on
+         * the gecko thread, so the dereference is a guaranteed NPE. Seed it with the
+         * main handler; the only runnable ever posted through this path
+         * (Clipboard.updateSequenceNumber) is main-thread safe, and native overwrites
+         * the field with the real gecko-thread handler once the thread boots.
+         */
+        internal fun ensureGeckoHandlerSeeded() {
+            try {
+                if (org.mozilla.gecko.util.ThreadUtils.sGeckoHandler == null) {
+                    org.mozilla.gecko.util.ThreadUtils.sGeckoHandler =
+                        android.os.Handler(android.os.Looper.getMainLooper())
+                    AppLogger.i(TAG, "Seeded ThreadUtils.sGeckoHandler with main handler (lifecycle race workaround)")
+                }
+            } catch (t: Throwable) {
+                AppLogger.w(TAG, "sGeckoHandler seed failed: ${t.message}")
+            }
+        }
+
         private fun createRuntime(context: Context): GeckoRuntime {
+            ensureGeckoHandlerSeeded()
             val settingsBuilder = GeckoRuntimeSettings.Builder()
                 .javaScriptEnabled(true)
                 .consoleOutput(true)

--- a/app/src/test/java/com/webtoapp/core/engine/GeckoViewEngineSeedTest.kt
+++ b/app/src/test/java/com/webtoapp/core/engine/GeckoViewEngineSeedTest.kt
@@ -1,0 +1,55 @@
+package com.webtoapp.core.engine
+
+import android.os.Handler
+import android.os.Looper
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertSame
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class GeckoViewEngineSeedTest {
+
+    @After
+    fun restoreHandler() {
+        org.mozilla.gecko.util.ThreadUtils.sGeckoHandler = null
+    }
+
+    @Test
+    fun `seed installs main handler when sGeckoHandler is null`() {
+        org.mozilla.gecko.util.ThreadUtils.sGeckoHandler = null
+
+        GeckoViewEngine.ensureGeckoHandlerSeeded()
+
+        val handler = org.mozilla.gecko.util.ThreadUtils.sGeckoHandler
+        assertNotNull(handler)
+        assertEquals(Looper.getMainLooper(), handler!!.looper)
+    }
+
+    @Test
+    fun `seed keeps an already-assigned handler untouched`() {
+        val preset = Handler(Looper.getMainLooper())
+        org.mozilla.gecko.util.ThreadUtils.sGeckoHandler = preset
+
+        GeckoViewEngine.ensureGeckoHandlerSeeded()
+
+        assertSame(preset, org.mozilla.gecko.util.ThreadUtils.sGeckoHandler)
+    }
+
+    @Test
+    fun `seed is idempotent and postable before native assignment`() {
+        org.mozilla.gecko.util.ThreadUtils.sGeckoHandler = null
+
+        GeckoViewEngine.ensureGeckoHandlerSeeded()
+        val first = org.mozilla.gecko.util.ThreadUtils.sGeckoHandler
+        GeckoViewEngine.ensureGeckoHandlerSeeded()
+
+        assertSame(first, org.mozilla.gecko.util.ThreadUtils.sGeckoHandler)
+
+        // The ON_RESUME dispatch does exactly this before native boots; must not throw.
+        org.mozilla.gecko.util.ThreadUtils.sGeckoHandler!!.post {}
+    }
+}

